src/legacy_handler.rs

```diff
@@ -298,6 +315,49 @@ impl LegacyHandler {
             serializer.append_pair("node3", on_bit);
             serializer.append_pair("node4", on_bit);
         }
+        self.response_printer = Some(result_printer);
         Ok(())
     }
 }
+
+fn print_power_status_nodes(map: &serde_json::Value) -> anyhow::Result<()> {
+    let results = map
+        .get("result")
+        .context("api error")?
+        .as_array()
+        .context("api error")?[0]
+        .as_object()
+        .context("response parse error")?;
+
+    for (key, value) in results {
+        let number = value.as_str().context("api error")?.parse::<u8>()?;
+        let status = if number == 1 { "On" } else { "off" };
+        println!("{}: {}", key, status);
+    }
+
+    Ok(())
+}
+
+fn result_printer(result: &serde_json::Value) -> anyhow::Result<()> {
+    let res = result.get("result").context("api error")?;
+    println!("{}", res.as_str().context("api error")?);
+    Ok(())
+}
+
+fn print_usb_status(map: &serde_json::Value) -> anyhow::Result<()> {
+    let results = map
+        .get("result")
+        .context("api error")?
+        .as_array()
+        .context("api error")?[0]
+        .as_object()
+        .context("response parse error")?;
+
+    println!(
+        "Usb bus is routed to {} and acting as a USB {}",
+        results["node"].as_str().unwrap().to_lowercase(),
```

Comment:
Instead of unwrapping and panicking this should leave early with a user-friendly message